_posts/2020-04-01-database-migration-spring-boot-liquibase.md

```diff
@@ -0,0 +1,304 @@
+</div> 
+
+Now that we added our first migration when Spring Boot application is run, it runs liquibase during the startup. Liquibase executes the changeSet which creates the `user_details` table with `user_pkey` as the primary key.
+
+### Usage of Changelog parameters:
+
+ Changelog parameters come in very handy when we want to abstract differences between environments while creating changelogs. These can be set using application property `spring.liquibase.paramaters`, which takes a map of key and value pairs. Below code demonstrates the usage of changelog parameters to select different data types while executing a changeSet in different environments.
+
+The following application property file sets a liquibase parameter `textColumnType`. Spring profile `h2` defines
+`VARCHAR(250)` as `textColumnType` and `docker` profile defines `TEXT` as `textColumnType`.
+
+```yaml
+spring:
+  profiles: docker
+  liquibase:
+    parameters:
+      textColumnType: TEXT
+    contexts: local
+---
+spring:
+  profiles: h2
+  liquibase:
+    parameters:
+      textColumnType: VARCHAR(250)
+    contexts: local    
+```
+The changelog file below uses the liquibase parameter `textColumnType` to define the column type. When the application runs with `docker` profile it uses `TEXT` as the type and for `h2` profile it uses `VARCHAR(250)`.
+
+```yaml
+databaseChangeLog:
+  - changeSet:
+     ...
+      changes:
+        - createTable:
+            columns:
+             ...
+              - column:
+                  constraints:
+                    nullable: false
+                  name: username
+                  type: ${textColumnType}
+```
+
+### Usage of Liquibase context:
+ 
+ As described earlier, context can be used to control which changeSets need to be run. The code example demonstrates the use of context in a changeSet, which is used to add test data load changeSet in a non-prod environment.
+
+Below changeSet has context expression `!prod` with this the changeSet will be run for all contexts expect `prod`.
+
+ ```xml
+<databaseChangeLog>
+  <changeSet author="liquibase-docs" id="loadUpdateData-example" context="!prod">
+    <loadUpdateData
+      encoding="UTF-8"
+      file="db/data/users.csv"
+      onlyUpdate="false"
+      primaryKey="id"
+      quotchar="'"
+      separator=","
+      tableName="user_details">
+    </loadUpdateData>
+  </changeSet>
+</databaseChangeLog>
+```
+Context values can be passed to liquibase using the property `spring.liquibase.contexts` as shown below 
+
+```yaml
+---
+spring:
+  profiles: docker
+  liquibase:
+    parameters:
+      textColumnType: TEXT
+    contexts: nonprod
+```
+
+
+### Configure Liquibase in Spring Boot
+
+Below is the list of all properties that Spring Boot provides to configure the behavior of Liquibase.
+
+```yaml
```

Comment:
I like the idea of explaining all configuration parameters a little better than the Spring Boot docs. Please change this into a table with the columns "property" and "description", though, because that would be much better readable. You can find an example with a table here: https://github.com/reflectoring/reflectoring.github.io/blob/master/_posts/2019-07-24-maven-scopes-gradle-configurations.md

Reply:
done